### ISSUE TITLE
GKD: multimodal inputs?

### ISSUE THREAD

**Issue by e-zorzi** [huggingface/trl#4368]
Does the Generalized Knowledge Distillation trainer (GKDTrainer) support multimodal inputs (VLMs)?
If yes, what's the expected dataset format? There is no example of this in the documentation.

Thanks!

**Comment by e-zorzi** [comment]
I understand the problem is that GKDTrainer does not support a custom DataCollator

    class GKDTrainer(SFTTrainer):
        def __init__(
            ...
            data_collator: Optional[DataCollator] = None,  # type: ignore
            ...
        ):
            data_collator = DataCollatorForChatML(tokenizer=processing_class, max_length=args.max_length)
            ...
   
By reading around, GRPOTrainer had a similar problem before supporting multimodal inputs #2807, and people did try to patch in a custom collator, but @qgallouedec said that that was not the way to go and that one should instead change the generation and loss computation parts.

However, GKDTrainer is much more similar to SFTrainer than GRPOTrainer, so implementing a new collator seems the way to go in this case. 
So my question is: if I add a custom collator and then make sure that the generation and loss parts work correctly (in particular, that they work with the right tokenized input), this should work, right? It would be much easier than changing GRPOTrainer, am I wrong?

**Comment by e-zorzi** [comment]
Actually, I might have had an older version of TRL. Version 0.24.0 seems to support custom collators.
